File: java/lang/src/main/java/otterop/lang/ArrayIterator.java
package otterop.lang;

class ArrayIterator<T> implements OOPIterator<T> {

    private Array<T> array;
    private int i;

    ArrayIterator(Array<T> array) {
        this.array = array;
        this.i = 0;
    }

    public boolean hasNext() {
        return this.i < this.array.size();
    }

    public T next() {
        T ret = this.array.get(this.i);
        this.i++;
        return ret;
    }
}

File: dotnet/src/Lang/Otterop/Lang/ArrayIterator.cs
namespace Otterop.Lang
{
    class ArrayIterator<T> : OOPIterator<T>
    {
        private Array<T> array;
        private int i;
        internal ArrayIterator(Array<T> array)
        {
            this.array = array;
            this.i = 0;
        }

        public bool HasNext()
        {
            return this.i < this.array.Size();
        }

        public T Next()
        {
            T ret = this.array.Get(this.i);
            this.i++;
            return ret;
        }

    }

}
